### ISSUE TITLE
When number of out episodes is known never set detected episode number higher.

### ISSUE THREAD

**Issue by kmaid** [MALSync/MALSync#624]
This might be a bug.

Sometimes sources don't number episodes by season but by number of episodes. An example of this is

Episode Count: https://animepahe.com/anime/7739b0f5-bb1d-27a0-c2a3-f3f99a90c1c5
Season & Episode Count:  https://gogoanime.ai/category/tensei-shitara-slime-datta-ken-2nd-season

Currently when you watch the first episode of the second season on animepahe it will set the second season to complete even if only one episode is out. 

Proposal: When episode number detected is higher than known number of episodes out set to known number of episodes out. This will at least help people who are keeping up to date with a series.

**Comment by farahnur42** [comment]
@kmaid If I'm not reading this incorrectly then what you're looking for is the correction menu (default hotkey is C for it), change out the MAL entry there to the *correct* season entry and then add an episode offset (to the tunes of offsetting out the episode count *before* the specific season started). There isn't any good solution for this that can be reasonably automated so long as *some* sites put all the seasons in one page and others don't + MAL, AL, Kitsu all explicitly have separate season entries per title.

**Comment by kmaid** [comment]
What about using the notification to highlight there has probably been an episode tracking issue and offering the correction menu? Then you would quickly crowd source the data.

**Comment by farahnur42** [comment]
Episode offsets are not crowd-sourced as far as I know (they are a workaround that exist to make sure MALSync still works for the handful of sites with these sorts of issues), the correction menu is primarily there to facilitate fixing incorrect mappings (for when say MALSync detects the wrong title) and you can *optionally* send in the correction to the external database.

**Comment by kmaid** [comment]
Warning the user that the episode tracking has probably gone wrong would still be helpful

**Comment by farahnur42** [comment]
Yeah that I do agree with, however I'm not one of the MALSync devs so one of them will have to chime in and figure that out.

**Comment by lolamtisch** [comment]
Soon malsync will use taigas `anime-relations` list. This hopefully should fix most of the Problems automatically.
https://github.com/MALSync/MALSync/pull/620

**Comment by kmaid** [comment]
I think more rules will help but there will still be cases where we don't have the rules yet and could fail more gracefully.


Fwiw thank you for MAL sync i use it all the time 👍

**Comment by DexLuther** [comment]
> @kmaid If I'm not reading this incorrectly then what you're looking for is the correction menu (default hotkey is C for it), change out the MAL entry there to the _correct_ season entry and then add an episode offset (to the tunes of offsetting out the episode count _before_ the specific season started). There isn't any good solution for this that can be reasonably automated so long as _some_ sites put all the seasons in one page and others don't + MAL, AL, Kitsu all explicitly have separate season entries per title.

Where exactly is "C' supposed to be pressed? I've tried it everywhere inside of Malsync on the MAL site, on the anime's page on the MAL site, on Anilist, etc. Nothing happens.

**Comment by lolamtisch** [comment]
@DexLuther https://github.com/MALSync/MALSync/wiki/Troubleshooting#myanimeentry--anilist--kitsu--simkl-entry-is-not-correct

**Comment by DexLuther** [comment]
> @DexLuther https://github.com/MALSync/MALSync/wiki/Troubleshooting#myanimeentry--anilist--kitsu--simkl-entry-is-not-correct

Except I don't use Crunchyroll. I guess people who use Jellyfin/Plex are out of luck since you can't seem to do it from inside the extension (like clicking on the item in the List Sync menu).

**Comment by DexLuther** [comment]
> @DexLuther that works on every page and on the Videoplayer page too

Except it doesn't. 
I press "C" on the MAL page for a show that isn't synced properly. Nothing happens.
I press "C" on the Anilist page for a show that isn't synced properly. Nothing happens.
I press "C" on either of those pages while having the extension open. Nothing happens.
I press "C" on the Jellyfin WebUI for a show that isn't synced properly. Nothing happens.
I press "C" while playing an episode that isn't being synced properly. Nothing Happens.
I press "C" while playing an episode that isn't being synced properly while having MALSync open. Nothing happens.

I try all of that again while having MALSync open and on the List Sync showing the item to "update", but pressing "C" still does nothing.

I don't know what "Works on every page" means, but I've tried it on all the pages I would assume it should work, but doesn't. I've never got it to work on Chrome, and now I can't get it to work on Firefox either.

The only place I've gotten it to work is going to GoAnime, but like I said before shouldn't really have to go there to fix sync issues. Having that box show up on the MAL, AL, or Kitsu pages would make a lot more sense, and/or being able to just click on the item in the Sync list to fix issues.

Also, If you look at https://github.com/RickDB/PlexAniSync/ more precisely https://github.com/RickDB/PlexAniSync-Custom-Mappings/ and how they resolve these issues. MALsync could do something similar and depend largely on the community to keep the lists updated. 

I don't know if extensions can regularly (just before performing a sync) pull from Github to update the list. I assume they can because that's kinda what a lot of YouTube "utility" extensions pull much more information from servers or other places. AdBlocker extensions also regularly update their lists, so MALSync should be able to.

**Comment by farahnur42** [comment]
@DexLuther Just FYI that's a configurable shortcut, sometimes C is reserved and you can change it from Settings -> Tracking 
![image](https://github.com/MALSync/MALSync/assets/37258115/297ca670-cc78-405d-a653-e558519b7f3a). The correction menu is only intended to be opened from with either the overview page or the episode page, it won't do anything while you're on a MAL/AniList tab. Additionally, MALSync already piggybacks off of https://github.com/erengy/anime-relations, although I don't know if it is applied for Jellyfin/Plex as well off the top of my head.

**Comment by DexLuther** [comment]
Sure. Set the button to whatever you want. Then press it on a MAL, Anilist
page about any anime. You know where it would actually make sense. Now
press the button you set. Notice how nothing happens.

Jobless Reincarnation is one example. MALsync constantly wants to sync 11
episodes to 23, but fails. Despite both sites having it as two parts (11
and 12 episodes).

You'd think that clicking the box in the list sync or pressing the
correction button you set on the Anilist or MAL pages would allow you to
correct  it, but that would be too logical.

Instead you have to go to a pirate site (that I have to constantly accept
new permissions for despite not using) or presumably Crunchyroll to do it.
